### ISSUE TITLE
Nesting

### ISSUE THREAD

**Issue by jessicagood** [StraboSpot/StraboSpot-Mobile#155]
@nicolasmroberts Can you get us a UI mockup for Nesting? Thanks!

**Comment by nicolasmroberts** [comment]
Here is the basic layout. "Child" spots are organized by generation. Within each generation, they are organized by their different image basemaps and (for polygons) spots that are geographically included. "Parent" spots are drawn with vertical lines. In both cases, spots that have sunspots within them are labeled with "[x subspots]"

<img width="417" alt="Screen Shot 2020-09-17 at 12 34 31 PM" src="https://user-images.githubusercontent.com/35352527/93528033-5be05980-f8ff-11ea-96dc-bfdcc0dd7b40.png">
<img width="418" alt="Screen Shot 2020-09-17 at 12 34 17 PM" src="https://user-images.githubusercontent.com/35352527/93528038-5c78f000-f8ff-11ea-8038-25e46f3a32a0.png">
<img width="432" alt="Screen Shot 2020-09-17 at 12 32 42 PM" src="https://user-images.githubusercontent.com/35352527/93528039-5c78f000-f8ff-11ea-86c5-f0d3a7c49874.png">

**Comment by nicolasmroberts** [comment]
and ignore the weird dark bars between the little icons -- this should appear just like the little icons in the spots list page

**Comment by jessicagood** [comment]
Added with 234bfb5c420b400ab7b25517cba039881e284327 and a0bf0d3349f60597561ae2203794f07bb1167342.

Still more UI work to do.

@nicolasmroberts Not sure about showing all of the subspots for each Spot listed in the nest. We can do it but if there are very many Spots that becomes a lot of calculations and performance would likely suffer.

**Comment by nicolasmroberts** [comment]
@jessicagood I think it would be fine to stop at 2 generations of children. Can you still add the number of sub spots within each spot so the user has a sense of where to burrow down into?

**Comment by jessicagood** [comment]
As I did in SS1, I have it set to display up to 10 generations of parents and 10 generations of children. What I was referring to in my previous post was calculating subspots of each of the parents in each parent generation. Only having 2 generations of parents may cut down a bit on calculations but often the nests aren't too deep anyway so then again it may not. It just becomes an issue (maybe) when there are a lot of Spots since every Spot has to be checked to see if it's a parent or a child.

For now I still have it at 10 generations and I added the subspots (commit 9788736d8954ffe4d166ab8d23bd8d3d8dcda679). Let's see how that runs in a test project with a lot of Spots and we can always easily cut the number of generations back and/or not show subspots.

The UI is slightly different than what you have above so it better matches the style of the Spots list in the main menu. See how you like it or if you want changes. I still need to add the image basemap thumbnails though.
